### PR TITLE
[mono] Implement Interop.Libraries.cs for iOS

### LINKED ISSUE #34637: Dynamic libraries in the microsoft.netcore.app.runtime.ios-x64 package has incorrect identity
Repro:

```shell
otool -l ~/.nuget/packages/microsoft.netcore.app.runtime.ios-x64/5.0.0-*/runtimes/ios-x64/native/*dylib | grep LC_ID_DYLIB -A 2
          cmd LC_ID_DYLIB
      cmdsize 184
         name /Users/runner/runners/2.165.2/work/1/s/artifacts/obj/native/netcoreapp5.0-iOS-Release-x64/System.IO.Compression.Native/libSystem.IO.Compression.Native.dylib (offset 24)
```

This should be something like `@executable_path/libSystem.IO.Compression.Native.dylib`.

CC @akoeplinger

### PR DESCRIPTION
**Pull request by EgorBo**
Fixes https://github.com/dotnet/runtime/issues/34637

@rolfbjarne did you mean something like this?

### CODE REVIEW

**Comment by rolfbjarne** [comment]
No, to fix #34637 you need to pass `-Wl,-install_name,@executable_path/libfoo.dylib` to clang when compiling the dylib (or alternatively use `install_name_tool` after the dylib has been built).

File: src/mono/netcore/System.Private.CoreLib/src/System/Interop.Libraries.iOS.cs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+internal static partial class Interop
+{
+    internal static partial class Libraries
+    {
+        internal const string SystemNative = "@executable_path/libSystem.Native.dylib";
+        internal const string GlobalizationNative = "@executable_path/libSystem.Globalization.Native.dylib";
+        internal const string NetSecurityNative = "@executable_path/libSystem.Net.Security.Native.dylib";
+        internal const string CryptoNative = "@executable_path/libSystem.Security.Cryptography.Native.OpenSsl.dylib";
+        internal const string CompressionNative = "@executable_path/libSystem.IO.Compression.Native.dylib";
+        internal const string IOPortsNative = "@executable_path/libSystem.IO.Ports.Native.dylib";
```
Comment (stephentoub) [inline src/mono/netcore/System.Private.CoreLib/src/System/Interop.Libraries.iOS.cs:14]: Is `@executable_path` somehow recognized by the tool chain, e.g. the linker, and rewritten?  Or is that actually a directory special name recognized by iOS?

Comment (rolfbjarne) [reply src/mono/netcore/System.Private.CoreLib/src/System/Interop.Libraries.iOS.cs:14]: `@executable_path` is expanded to the directory of the main executable by the dynamic loader (dyld) on Apple platforms: https://www.mikeash.com/pyblog/friday-qa-2009-11-06-linking-and-install-names.html (when `@executable_path` is encountered when loading dynamic libraries).

Comment (stephentoub) [reply src/mono/netcore/System.Private.CoreLib/src/System/Interop.Libraries.iOS.cs:14]: Got it, thanks.

**Comment by akoeplinger** [comment]
As Rolf said this is not the correct way, we need to fix it on the cmake side. I tried a few things and couldn't get it to work, from reading cmake docs having https://github.com/dotnet/runtime/blob/174f13637c1f5b422fa01f5cf8ff3911bca2f3e0/src/libraries/Native/Unix/CMakeLists.txt#L17 should actually be enough but for some reason it's not.

**Pull request closed without merging**